### PR TITLE
Album remote operations

### PR DESCRIPTION
**Pull request by surinder-tsys**
This PR contains the Remote operations for the Albums functionality.

### CODE REVIEW

File: library/src/main/java/com/owncloud/android/lib/resources/albums/PhotoAlbumEntry.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    private fun parseLong(value: String?): Long {
+        return try {
+            value?.toLong() ?: 0L
+        } catch (e: NumberFormatException) {
+            0L
+        }
+    }
+
+    val albumName: String
```
Comment (alperozturk96) [inline library/src/main/java/com/owncloud/android/lib/resources/albums/PhotoAlbumEntry.kt:67]: Using methods like regular expressions and string manipulations (e.g., substring, split, etc.) can often be error prone, especially when dealing with edge cases. It would be beneficial to add tests to ensure correctness across a variety of scenarios.

Could you add tests and simplify the logic if possible?

Similar logic (not sure if it's fitting for current context)

```
val albumName: String
        get() {
            return href
                .removeSuffix("/")
                .substringAfterLast("/")
                .takeIf { it.isNotEmpty() } ?: ""
        }
```

**Review by alperozturk96 — CHANGES_REQUESTED** [review]

File: library/src/main/java/com/owncloud/android/lib/resources/albums/PhotoAlbumEntry.kt (hunk 1 above)
Comment (surinder-tsys) [reply library/src/main/java/com/owncloud/android/lib/resources/albums/PhotoAlbumEntry.kt:67]: Done as per comment and added test cases.

**Review by alperozturk96 — CHANGES_REQUESTED** [review]
Thank you for the update. You can use the following command to fix spot bugs and formatting.

`./gradlew detekt` → it detects and gives HTML output, so that you can check it.
`./gradlew :app:spotlessApply` → reformats Kotlin code

**Review by alperozturk96 — CHANGES_REQUESTED** [review]
The following issues were found during analysis. Please fix them where possible. Thank you.

**NestedBlockDepth** (You can suppress with annotation if you are not able to extract the nested blocks) → ReadAlbumItemsRemoteOperation.kt:28:18

**TooGenericExceptionCaught** (You can suppress with annotation since error handling in remote operation is not well-designed in the current architecture) →
→ CopyFileToAlbumRemoteOperation.kt:88:18
→ CreateNewAlbumRemoteOperation.kt:57:18
→ ReadAlbumItemsRemoteOperation.kt:68:18
→ ReadAlbumsRemoteOperation.kt:57:18
→ RemoveAlbumFileRemoteOperation.kt:45:18
→ RemoveAlbumRemoteOperation.kt:53:18
→ RenameAlbumRemoteOperation.kt:55:18

**MagicNumber** (You can create constants in companion object and use them):
→ CopyFileToAlbumRemoteOperation.kt:131:76
→ PhotoAlbumEntry.kt:83:61

**NewLineAtEndOfFile** (You can add a new line at the end of the file):
→ PhotoAlbumEntry.kt:91:2
→ ToggleAlbumFavoriteRemoteOperation.kt:73:2
→ PhotoAlbumEntryTest.kt:57:2

**ReturnCount** (You can try to reduce return counts. If not possible, you can use annotation to suppress it):
→ CopyFileToAlbumRemoteOperation.kt:42:18

**Comment by surinder-tsys** [comment]
@AndyScherzinger  @alperozturk96 I have taken care of copyright and updated PR.

**Comment by alperozturk96** [comment]
@surinder-tsys Thank you for the update. There are three remaining items to address, once those are resolved, I can proceed with reviewing the client PR. If everything looks good there, we can move forward with merging this PR.

###  check (spotlessKotlinCheck)

 src/main/java/com/owncloud/android/lib/resources/albums/CopyFileToAlbumRemoteOperation.kt:L1 ktlint(standard:max-line-length) Exceeded max line length (120)
 
### Spotbugs
 
Screenshot 2025-06-20 at 15 21 21

### DCO

Screenshot 2025-06-20 at 15 22 17

File: library/src/main/java/com/owncloud/android/lib/resources/albums/ToggleAlbumFavoriteRemoteOperation.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+class ToggleAlbumFavoriteRemoteOperation
+    @JvmOverloads
+    constructor(
+        private val makeItFavorited: Boolean,
+        private val filePath: String,
+        private val sessionTimeOut: SessionTimeOut = defaultSessionTimeOut
+    ) : RemoteOperation<Any>() {
+        @Deprecated("Deprecated in Java")
+        override fun run(client: OwnCloudClient): RemoteOperationResult<Any> {
+            var result: RemoteOperationResult<Any>
```
Comment (alperozturk96) [inline library/src/main/java/com/owncloud/android/lib/resources/albums/ToggleAlbumFavoriteRemoteOperation.kt]: `filePath` can be empty or blank. Better to check in the beginning of the function.

**Review by alperozturk96 — CHANGES_REQUESTED** [review]
Hey

Thank you for the PR. 👍 

I found a few minor things. Hopefully, once those are sorted out, we can test the client PR and then move ahead with the merge. 

Thanks again for the contribution :)

**Review by alperozturk96 — DISMISSED** [review]

**Review by alperozturk96 — APPROVED** [review]

**Review by alperozturk96 — CHANGES_REQUESTED** [review]
we should fix ci

**Comment by tobiasKaminsky** [comment]
Seems tests were only flaky ✔

**Review by tobiasKaminsky — DISMISSED** [review]

**Review by alperozturk96 — APPROVED** [review]
